presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -150,8 +154,18 @@ public PinotQueryGeneratorContext withAggregation(
             int aggregations,
             Set<VariableReferenceExpression> hiddenColumnSet)
     {
-        // there is only one aggregation supported.
-        checkSupported(!hasAggregation(), "Pinot doesn't support aggregation on top of the aggregated data");
+        // there is only one aggregation supported unless distinct is used.
+        boolean passed = false;
```

Comment:
Can you use a more descriptive name than passed, please?

Reply:
will do